src/utils/NovorAdapter.cpp

```diff
@@ -0,0 +1,389 @@
+    // thirdparty executable 
+    registerInputFile_("executable", "<jar>", "novor.jar", "novor.jar", false, false, ListUtils::create<String>("skipexists"));
+    // input and output
+    registerInputFile_("in", "<file>", "", "MzML Input file");
+    setValidFormats_("in", ListUtils::create<String>("mzml"));
+    registerOutputFile_("out", "<file>", "", "Novor idXML output");
+    setValidFormats_("out", ListUtils::create<String>("idXML"));
+    // enzyme
+    registerStringOption_("enzyme", "<choice>", "Trypsin", "Digestion enzyme - currently only Trypsin is supported ", false);
+    setValidStrings_("enzyme", ListUtils::create<String>("Trypsin"));
+    // instrument
+    registerStringOption_("fragmentation", "<choice>", "CID", "Fragmentation method", false);
+    setValidStrings_("fragmentation", ListUtils::create<String>("CID,HCD"));
+    registerStringOption_("massAnalyzer", "<choice>" , "Trap", "MassAnalyzer e.g. (Oritrap CID-Trap, CID-FT, HCD-FT; QTof CID-TOF)", false);
+    setValidStrings_("massAnalyzer", ListUtils::create<String>("Trap,TOF,FT"));
+    // mass error tolerance
+    registerDoubleOption_("fragmentIon_error_tolerance", "<double>", 0.5, "Fragmentation error tolerance  (Da)", false);
+    registerDoubleOption_("precursor_error_tolerance", "<double>" , 15.0, "Precursor error tolerance  (ppm or Da)", false);
+    registerStringOption_("precursor_error_units", "<choice>", "Da", "Unit of precursor mass tolerance", false);
+    setValidStrings_("precursor_error_units", ListUtils::create<String>("Da,ppm"));
+    // post-translational-modification
+    registerStringList_("variable_modifications", "<mods>", vector<String>(), "Variable modifications", false);
+    setValidStrings_("variable_modifications", ListUtils::create<String>("Acetyl (K),Acetyl (N-term),Amidated (C-term),Ammonia-loss (N-term C),Biotin (K),Biotin (N-term),Carbamidomethyl (C),Carbamyl (K),Carbamyl (N-term),Carboxymethyl (C),Deamidated (NQ),Dehydrated (N-term C),Dioxidation (M),Methyl (C-term),Methyl (DE),Oxidation (M),Oxidation (HW),Phospho (ST),Phospho (Y),Pyro-carbamidomethyl (N-term C),Pyro-Glu (E),Pyro-Glu (Q),Sodium (C-term),Sodium (DE),Sulfo (STY),Trimethyl (RK)"));
+    registerStringList_("fixed_modifications", "<mods>", vector<String>(), "Fixed modifications", false);
+    setValidStrings_("fixed_modifications", ListUtils::create<String>("Acetyl (K),Acetyl (N-term),Amidated (C-term),Ammonia-loss (N-term C),Biotin (K),Biotin (N-term),Carbamidomethyl (C),Carbamyl (K),Carbamyl (N-term),Carboxymethyl (C),Deamidated (NQ),Dehydrated (N-term C),Dioxidation (M),Methyl (C-term),Methyl (DE),Oxidation (M),Oxidation (HW),Phospho (ST),Phospho (Y),Pyro-carbamidomethyl (N-term C),Pyro-Glu (E),Pyro-Glu (Q),Sodium (C-term),Sodium (DE),Sulfo (STY),Trimethyl (RK)"));
+   // forbidden residues
+   registerStringList_("forbiddenResidues", "<mods>", vector<String>(), "Forbidden Resiudes", false);
+   setValidStrings_("forbiddenResidues", ListUtils::create<String>("I,U"));
```

Comment:
what about options not available through the adapter? is there a config file you can pass to the tool ? or are these all the available options?

Reply:
These are all the available options, but there seem to be some external options in the standard param.txt of novor. I will add the option to use the param.txt directly if needed (as suggested from you below - optional).